cli/azd/internal/cmd/add/add_configure.go

```diff
@@ -0,0 +1,209 @@
+package add
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"unicode"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/fatih/color"
+)
+
+type promptCtx struct {
+	context.Context
+	prj *project.ProjectConfig
+}
+
+// configure fills in the fields for a resource.
+func configure(
+	r *project.ResourceConfig,
+	console input.Console,
+	ctx promptCtx) (*project.ResourceConfig, error) {
+	switch r.Type {
+	case project.ResourceTypeHostContainerApp:
+		return fillUses(r, console, ctx)
+	case project.ResourceTypeOpenAiModel:
+		return fillAiModelName(r, console, ctx)
+	case project.ResourceTypeDbPostgres,
+		project.ResourceTypeDbMongo:
+		return fillDatabaseName(r, console, ctx)
+	case project.ResourceTypeDbRedis:
+		r.Name = "redis"
+		return r, nil
+	default:
+		return r, nil
+	}
+}
+
+func fillDatabaseName(
+	r *project.ResourceConfig,
+	console input.Console,
+	ctx promptCtx) (*project.ResourceConfig, error) {
+	if r.Name != "" {
+		return r, nil
+	}
+
+	for {
+		dbName, err := console.Prompt(ctx, input.ConsoleOptions{
+			Message: fmt.Sprintf("Input the name of the app database (%s)", r.Type.String()),
+			Help: "Hint: App database name\n\n" +
+				"Name of the database that the app connects to. " +
+				"This database will be created after running azd provision or azd up.",
+		})
+		if err != nil {
+			return r, err
+		}
+
+		if err := validateResourceName(dbName, ctx.prj); err != nil {
+			console.Message(ctx, err.Error())
+			continue
+		}
+
+		r.Name = dbName
+		break
+	}
+
+	return r, nil
+}
+
+func fillAiModelName(
+	r *project.ResourceConfig,
+	console input.Console,
+	ctx promptCtx) (*project.ResourceConfig, error) {
+	if r.Name != "" {
+		return r, nil
+	}
+
+	modelProps, ok := r.Props.(project.AIModelProps)
+	defaultName := ""
+
+	// provide a default suggestion using the underlying model name
+	if ok {
+		defaultName = modelProps.Model.Name
+		i := 1
+		for {
+			if _, exists := ctx.prj.Resources[defaultName]; exists {
+				i++
+				defaultName = fmt.Sprintf("%s-%d", defaultName, i)
+			} else {
+				break
+			}
+		}
+	}
+
+	for {
+		modelName, err := console.Prompt(ctx, input.ConsoleOptions{
+			Message:      "What is the name of this model?",
```

Comment:
Just wondering (maybe we'll discover this from user studies) - I assume this is the resource name of the model, not something like "gpt-3.5" right?  Do users understand what "name of this model" means here?